pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -302,15 +302,23 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error
 		return err
 	}
 	if pod == nil {
-		// Pod is not present, create pod.
 		pod, err = c.createPod(tr, rtr.TaskSpec, rtr.TaskName)
 		if err != nil {
-			// This Run has failed, so we need to mark it as failed and stop reconciling it
 			var reason, msg string
+			succeededStatus := corev1.ConditionUnknown
 			if isExceededResourceQuotaError(err) {
 				reason = reasonExceededResourceQuota
-				msg = getExceededResourcesMessage(tr)
+				count, deadline, isNewBackoff := c.timeoutHandler.GetBackoff(tr)
+				msg = fmt.Sprintf("%s, reattempted %d times", getExceededResourcesMessage(tr), count)
+				if isNewBackoff {
+					taskRunEndTime := reconciler.GetTaskRunEndTime(tr)
+					if deadline.After(taskRunEndTime) {
+						deadline = taskRunEndTime
```

Comment:
since `GetBackoff` takes the entire taskRun as a param (not that it should - maybe it shouldnt), I'm kind of surprised it doesn't know how to determine if the deadline is past the end of the execution time :thinking: